### PR TITLE
Convert Forms Demo to project.json

### PR DESCRIPTION
**Pull request by MatkovIvan**
Rename crashes assembly as well as other modules (329cc13).
The build scripts and nuspec are also updated.

### CODE REVIEW

**Comment by guperrot** [comment]
@achocron basically we can't hide APIs in UWP crashes if we want to support people using project.json...

**Comment by MatkovIvan** [comment]
Build fails on CI with error:
`The agent MCSDK1 lost communication with the server. Verify the machine is running and has a healthy network connection.`

**Comment by guperrot** [comment]
Restored feature/use-project-json branch deleted by mistake

**Comment by MatkovIvan** [comment]
Merged with https://github.com/Microsoft/mobile-center-sdk-dotnet/pull/267 (develop) and retarget to develop again.

**Comment by MatkovIvan** [comment]
Guys, let me know if anything should be changed here to move on.

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.iOS/project.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "dependencies": {
```
Comment (guperrot) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.iOS/project.json]: Missing push

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.Droid/project.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "dependencies": {
```
Comment (guperrot) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.Droid/project.json]: Missing push

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.Droid/Contoso.Forms.Demo.Droid.csproj [hunk 3]
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\..\..\packages\Xamarin.Build.Download.0.4.4\build\Xamarin.Build.Download.props" Condition="Exists('..\..\..\packages\Xamarin.Build.Download.0.4.4\build\Xamarin.Build.Download.props')" />
-  <Import Project="../../../packages/Xamarin.Build.Download.0.4.3/build/Xamarin.Build.Download.props" Condition="Exists('../../../packages/Xamarin.Build.Download.0.4.3/build/Xamarin.Build.Download.props')" />
+  <Import Project="../../../packages/Xamarin.Build.Download.0.4.4/build/Xamarin.Build.Download.props" Condition="Exists('../../../packages/Xamarin.Build.Download.0.4.4/build/Xamarin.Build.Download.props')" />
```
Comment (guperrot) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.Droid/Contoso.Forms.Demo.Droid.csproj]: This does not work as the package from another folder when using project.json. If git clean -xdf you will not be able to compile anymore as the package will be restored in ~/.nuget/packages.

We can get rid of this I guess.

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.iOS/Contoso.Forms.Demo.iOS.csproj [hunk 4]
```diff
@@ -134,6 +94,12 @@
     <Reference Include="Xamarin.Forms.Xaml">
       <HintPath>..\..\..\packages\Xamarin.Forms.2.3.4.231\lib\Xamarin.iOS10\Xamarin.Forms.Xaml.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Azure.Mobile.Push">
```
Comment (guperrot) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.iOS/Contoso.Forms.Demo.iOS.csproj]: wrong references to push, demo uses nuget.org now (0.11.1 has Xamarin push)

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.Droid/Contoso.Forms.Demo.Droid.csproj (hunk 3 above)
Comment (guperrot) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.Droid/Contoso.Forms.Demo.Droid.csproj]: Can remove this nuget package and this import.

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.Droid/Contoso.Forms.Demo.Droid.csproj [hunk 5]
```diff
@@ -258,38 +106,5 @@
     <Error Condition="!Exists('..\..\..\packages\Xamarin.Android.Support.v7.AppCompat.25.3.1\build\MonoAndroid70\Xamarin.Android.Support.v7.AppCompat.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\..\..\packages\Xamarin.Android.Support.v7.AppCompat.25.3.1\build\MonoAndroid70\Xamarin.Android.Support.v7.AppCompat.targets'))" />
```
Comment (guperrot) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.Droid/Contoso.Forms.Demo.Droid.csproj]: Remove all `<Error` tags referencing `../../packages`, this does not work anymore.

**Comment by guperrot** [comment]
Testing xamarin again with clean state.

**Pull request merged**